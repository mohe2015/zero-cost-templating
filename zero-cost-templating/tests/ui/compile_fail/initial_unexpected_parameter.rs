#![feature(async_closure, async_iterator, coroutines, gen_blocks, noop_waker)]

extern crate alloc;

use alloc::borrow::Cow;

use zero_cost_templating::yields;
use zero_cost_templating_macros::template_stream;

#[template_stream("test.html.hbs")]
pub async gen fn test() -> Cow<'static, str> {
    let template = yields!(test0("unexpected"));
}

pub fn main() {}
