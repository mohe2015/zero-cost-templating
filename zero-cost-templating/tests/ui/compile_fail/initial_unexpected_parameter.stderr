error[E0061]: this function takes 0 arguments but 1 argument was supplied
  --> tests/ui/compile_fail/initial_unexpected_parameter.rs:12:28
   |
12 |     let template = yields!(test_initial0("unexpected"));
   |                            ^^^^^^^^^^^^^ ------------
   |                                          |
   |                                          unexpected argument of type `&'static str`
   |                                          help: remove the extra argument
   |
note: function defined here
  --> tests/ui/compile_fail/initial_unexpected_parameter.rs:10:1
   |
10 | #[template_stream("test.html.hbs")]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   = note: this error originates in the attribute macro `template_stream` (in Nightly builds, run with -Z macro-backtrace for more info)
